subprojects/docs/src/docs/release/notes.md

```diff
@@ -46,12 +46,16 @@ Native compile and link tasks now execute in parallel by default, making native
 
 The `zipTree` and `tarTree` implementations had a major performance issue, unpacking files every time the tree was traversed. This has now been fixed and should speed up builds using these trees a lot.
 
-### Better support for script plugins loaded via HTTP
+### Better support for script plugins from HTTP/HTTPS URLs
 
-Script plugins are applied to Gradle settings or projects via the `apply from: 'URL'` syntax. Support for `http://` and `https://` URLs has been improved in this release:
+Script plugins are applied to Gradle settings or projects with `apply from: 'URL'`. 
 
-- HTTP script plugins are cached for [`--offline`](userguide/dependency_management.html#sub:cache_offline) use.
-- Download of HTTP script plugins honours [proxy authentication settings](userguide/build_environment.html#sec:accessing_the_web_via_a_proxy).
+Support for script plugins from `http://` and `https://` URLs has been improved:
+
+- Script plugins are cached and only downloaded when necessary instead of on every build.
+- When using [`--offline`](userguide/dependency_management.html#sub:cache_offline), Gradle will not try to download script plugins if a cached copy exists.
+- [Proxy authentication settings](userguide/build_environment.html#sec:accessing_the_web_via_a_proxy) are now honored when downloading script plugins.
+- **NOTE**: URLs that contain a query component (information following a `?` in a URL) will not be cached. 
```

Comment:
I think it would make sense to cover this scenario as well at least as a follow up issue. Can you open one? From a user's perspective it will be kind of hard to figure out whether the script plugin was cached or not. I think we could indicate that to the user (in some form)

- Rendering a message on the console (at the very least on INFO log level).
- Clearly state the behavior in the user guide.
- We'll also need to adjust the training material to say that it is cached now. Let's create an issue for it and assign it to the next person delivering a training to fix the slides.

Reply:
I agree with you that it should be better.  I think the time would be time better spent just fixing the query string problem and reusing as much of the resource infrastructure for URIs that are not cached now. That way we're not creating an ad-hoc download message now. We never included this kind of information before, so we can make it better in 4.3.

I opened new issues (for the training material and for handling query strings).  #2925